### PR TITLE
Fix DataFrame slicing issues for empty cases

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Closes https://github.com/rapidsai/cudf/issues/10292

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py
Comment (vyasr) [inline python/cudf/cudf/core/dataframe.py:1244]: I assume this is to mimic pandas behavior?

File: python/cudf/cudf/tests/test_indexing.py
Comment (vyasr) [inline python/cudf/cudf/tests/test_indexing.py:1308]: I assume the size changes are unrelated to the fix, just to speed up the tests. Do they even need to be as big as they are? Would a few 10s or 100s of rows be enough?

**Review by vyasr — APPROVED** [review]
A couple small questions but this LGTM once you fix style and merge in the latest changes on 22.04. If you're using our pre-commit hooks I recommend rerunning pre-commit install once you merge in the latest 22.04 since I added a hook for the copyright checks now that we check those in CI.

File: python/cudf/cudf/core/dataframe.py
Comment (brandon-b-miller) [reply python/cudf/cudf/core/dataframe.py]: This is to match pandas yes. I actually had the rule slightly wrong here. The rule appears to be that when returning an empty slice from a `RangeIndex`'ed dataframe, a length zero range index is produced that spans from `start` to `stop`. 

```python
>>> df = pd.DataFrame({'a':[1,2,3], 'b':[4,5,6]})
>>> df[3:2:2].index
RangeIndex(start=3, stop=2, step=2)
```

However other kinds of index just get returned as an empty version of the original:

```python
>>> df
   a  b
1  1  4
2  2  5
4  3  6
>>> df[3:2:2].index
Int64Index([], dtype='int64')
```

**Pull request merged**
